tensorflow_addons/metrics/cohens_kappa.py

```diff
@@ -167,8 +167,9 @@ def result(self):
         # 6. Calculate Kappa score
         numerator = tf.reduce_sum(conf_mtx * weight_mtx)
         denominator = tf.reduce_sum(out_prod * weight_mtx)
-        kp = 1 - (numerator / denominator)
-        return kp
+        return tf.cond(tf.math.is_nan(denominator),
+                       true_fn=lambda: 0.0,
+                       false_fn=lambda: 1 - (numerator / denominator))
```

Comment:
Though I think the fix is correct, but a better approach should be change the division of [these two lines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/cohens_kappa.py#L161-L162) into `tf.math.divide_no_nan`. And modify `kp` to `kp = tf.math.divide_no_nan(denominator - numerator, denominator)`. How do you think about this @n3011?

Reply:
@WindQAQ there are two problems with `tf.math.divide_no_nan` approach:
   - it will produce `nan` result only, as `nan/nan` not defined.
   - it slightly slower than `tf.cond` based aproach.

Reply:
Got it. After a rough test, your approach is actually faster :-) Thanks for the report.